src/cascadia/UIHelpers/IconPathConverter.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include "IconPathConverter.g.h"
+
+namespace winrt::Microsoft::Terminal::UI::implementation
+{
+    struct IconPathConverter
+    {
+        IconPathConverter() = default;
+
+        static Windows::UI::Xaml::Controls::IconElement IconWUX(const winrt::hstring& iconPath);

```

Comment:
there is a difference between IconWUX and IconSourceWUX in that one of them supports `foo.dll,3` and the other does not. Which one does is left as an exercise for the reader. Why it does that is left as an exercise for @PankajBhojwani and @zadjii-msft.